### ISSUE TITLE
Highlighting quirks / suggestions

### ISSUE THREAD

**Issue by zhibog** [Platin21/odin.tmbundle#1]
**Quirks:**
- The arrow notation to mark a return value is properly highlighted, except when you return an array. In that case only the '-' gets highlighted. Slices work just fine.
- 'len' is always highlighted, even when it is part of a name
- ':' is highlighted on imports, but I suspect this is on purpose
- Weird highlighting when you have some constants within a struct. I don't know what the highlighting is supposed to look like here.

**Suggestions:**
- Since '+=' etc is highlighted, maybe you want '<<', '>>', '%', too, maybe even logical operators?
- Since 'len' is highlighted as a builtin, might want to add things like 'copy' as well and things like '#no_bounds_check'
- Add some sort of highlight to '@note' / '@todo' within comments. Really useful when looking through a codebase.

Now some images for clarification of the quirks in order of mentioning above:

![image](https://user-images.githubusercontent.com/48834794/60401319-d8164200-9b7f-11e9-8a02-c291616c5b1d.png)
![image](https://user-images.githubusercontent.com/48834794/60401354-7c988400-9b80-11e9-852f-2107559d39b7.png)
![image](https://user-images.githubusercontent.com/48834794/60401369-9a65e900-9b80-11e9-8cf3-c67893fec4bb.png)
![image](https://user-images.githubusercontent.com/48834794/60401376-b8cbe480-9b80-11e9-924f-b47afde37b36.png)

**Comment by Platin21** [comment]
Yeah the double point in imports is on purpose to separate the collection from the package. 

@NOTE and @TODO are both in there i might add the case insensitive version of them…

Looks like operators are borked on some places. 

For the structs yeah it currently doesn't like operators and such that needs some recursive stuff. 
On which i need to get my head around. 

yeah for #bounds_check and copy i can do builtin's too not that hard. 
The thing is one needs to know about them :)

Ahh the len stuff again yeah i need to get it done properly just thought highlighting the name alone would be enough as it's on boundary.
